impl/src/main/java/org/ehcache/spi/serialization/DefaultSerializationProvider.java

```diff
@@ -158,12 +158,22 @@ private AbstractProvider(Map<Class<?>, Class<? extends Serializer<?>>> serialize
     @Override
     public <T> Serializer<T> createKeySerializer(Class<T> clazz, ClassLoader classLoader, ServiceConfiguration<?>... configs) throws UnsupportedTypeException {
       DefaultSerializerConfiguration<T> conf = find(SerializerConfiguration.Type.KEY, configs);
+      if(conf != null) {
```

Comment:
Changes done in the `ClassInstanceProvider` class for issue 432 is preventing me from doing this refactoring. I have three choices here:

1. Leave this code as it is as an exceptional case
2. Revert some of the changes done in 432 (the changes removing `ConstructorArguments parameters from the _newInstance()_ methods in `ClassInstanceProvider` class)
3. Add a setter to `ClassInstanceConfiguration` that can be used to modify the *arguments* field. Using that setter I'll be able to inject the constructor args that I want for instantiation.

I personally prefer the third one. Please let me know if there are any objections/comments/feedback.